lib/enigma.rb

```diff
@@ -1,2 +1,121 @@
 class Enigma
+  attr_reader :characters
+  def initialize
+    @characters = ("a".."z").to_a << " "
+  end
+
+  def generate_key_string
+    rand(99999).to_s.rjust(5, "0")
+  end
+
+  def generate_keys(key_string)
```

Comment:
This method generates a the hash with the initial keys before adding the offsets. This will be extracted to a `Shift` class.